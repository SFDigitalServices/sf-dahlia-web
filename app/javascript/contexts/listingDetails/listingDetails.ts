interface ListingDetails {
  /**  */
  uid: string

  /**  */
  email: string

  /**  */
  created_at: Date

  /**  */
  updated_at: Date
}

interface ListingDetailsData {
  data: any
  success: boolean
}

export { ListingDetails, ListingDetailsData }
