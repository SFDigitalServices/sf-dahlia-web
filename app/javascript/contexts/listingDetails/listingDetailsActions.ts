import { createAction } from "typesafe-actions"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart } from "../../api/types/rails/listings/RailsAmiChart"

export enum ListingDetailsActions {
  StartFetchingUnits = "SetFetchingUnits",
  FinishFetchingUnits = "FinishFetchingUnits",
  SetUnits = "SetUnits",
  SetAmiCharts = "SetAmiCharts",
  StartFetchingAmiCharts = "SetFetchingAmiCharts",
  SetFetchedAmiCharts = "SetFetchedAmiCharts",
  SetFetchingAmiChartsError = "SetFetchingAmiChartsError",
  SetFetchingUnitsError = "SetFetchingUnitsError",
  FinishFetchingAmiCharts = "FinishFetchingAmiCharts",
}

export const startFetchingUnits = createAction(ListingDetailsActions.StartFetchingUnits)()
export const setUnits = createAction(ListingDetailsActions.SetUnits)()
export const setAmiCharts = createAction(ListingDetailsActions.SetAmiCharts)()
export const finishFetchingAmiCharts = createAction(ListingDetailsActions.FinishFetchingAmiCharts)<
  RailsAmiChart[]
>()
export const setFetchedAmiCharts = createAction(ListingDetailsActions.SetFetchedAmiCharts)()
export const startFetchingAmiCharts = createAction(ListingDetailsActions.StartFetchingAmiCharts)()
export const finishFetchingUnits = createAction(ListingDetailsActions.FinishFetchingUnits)<
  RailsUnit[]
>()
export const setFetchingUnitsError = createAction(
  ListingDetailsActions.SetFetchingUnitsError
)<Error>()
export const setFetchingAmiChartsError = createAction(
  ListingDetailsActions.SetFetchingAmiChartsError
)<Error>()
