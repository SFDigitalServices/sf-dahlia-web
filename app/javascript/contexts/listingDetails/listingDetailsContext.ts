import { createContext } from "react"
import RailsUnit from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart, RailsAmiChartMetaData } from "../../api/types/rails/listings/RailsAmiChart"

export type ContextProps = {
  units: RailsUnit[]
  amiCharts: RailsAmiChart[]
  fetchedAmiCharts: boolean
  fetchingAmiCharts: boolean
  fetchedUnits: boolean
  fetchingUnits: boolean
  fetchUnits: (listingId: string) => void
  fetchAmiCharts: (chartsToFetch: RailsAmiChartMetaData[]) => void
  fetchingAmiChartsError: Error
  fetchingUnitsError: Error
}

export default createContext<Partial<ContextProps>>({})
