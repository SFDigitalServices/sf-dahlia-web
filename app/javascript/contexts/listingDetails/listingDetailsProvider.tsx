import React, { useReducer } from "react"
import ListingDetailsReducer from "./listingDetailsReducer"
import ListingDetailsContext, { ContextProps } from "./listingDetailsContext"
import { getUnits, getAmiCharts } from "../../api/listingApiService"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart, RailsAmiChartMetaData } from "../../api/types/rails/listings/RailsAmiChart"
import {
  finishFetchingUnits,
  finishFetchingAmiCharts,
  startFetchingAmiCharts,
  startFetchingUnits,
  setFetchingAmiChartsError,
  setFetchingUnitsError,
} from "./listingDetailsActions"

interface ListingDetailsProviderProps {
  children?: React.ReactNode
}

const ListingDetailsProvider = (props: ListingDetailsProviderProps) => {
  const [state, dispatch] = useReducer(ListingDetailsReducer, {
    units: [],
    amiCharts: [],
    fetchingUnits: false,
    fetchedUnits: false,
    fetchingAmiCharts: false,
    fetchedAmiCharts: false,
    fetchingAmiChartsError: null,
    fetchingUnitsError: null,
  })

  const contextValues: ContextProps = {
    units: state.units,
    amiCharts: state.amiCharts,
    fetchedAmiCharts: state.fetchedAmiCharts,
    fetchingAmiCharts: state.fetchingAmiCharts,
    fetchedUnits: state.fetchedUnits,
    fetchingUnits: state.fetchingUnits,
    fetchingAmiChartsError: state.fetchingAmiChartsError,
    fetchingUnitsError: state.fetchingUnitsError,
    fetchUnits: (listingId: string) => {
      dispatch(startFetchingUnits())
      getUnits(listingId)
        .then((units: RailsUnit[]) => {
          dispatch(finishFetchingUnits(units))
        })
        .catch((error) => {
          console.error(error)
          dispatch(setFetchingUnitsError(error))
        })
    },
    fetchAmiCharts: (chartsToFetch: RailsAmiChartMetaData[]) => {
      dispatch(startFetchingAmiCharts())
      getAmiCharts(chartsToFetch)
        .then((amiCharts: RailsAmiChart[]) => {
          dispatch(finishFetchingAmiCharts(amiCharts))
        })
        .catch((error) => {
          console.error(error)
          dispatch(setFetchingAmiChartsError(error))
        })
    },
  }

  return (
    <ListingDetailsContext.Provider value={contextValues}>
      {props.children}
    </ListingDetailsContext.Provider>
  )
}

export default ListingDetailsProvider
