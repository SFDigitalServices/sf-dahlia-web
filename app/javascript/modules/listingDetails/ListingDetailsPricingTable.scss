.category-table {
  tr {
    td:before {
      font-size: var(--bloom-font-size-sm);

      @media (min-width: $screen-md) {
        font-size: var(--bloom-font-size-base);
      }
    }
    @media (min-width: $screen-md) {
      .stacked-table-header {
        &:first-child {
          width: 30%;
        }

        &:nth-child(2) {
          width: 40%;
        }

        &:last-child {
          width: 20%;
        }
      }
    }
  }
  .stacked-table-header {
    .stacked-table-cell-container {
      .stacked-table-cell {
        font-size: var(--bloom-font-size-sm);
        @media (min-width: $screen-md) {
          font-size: var(--bloom-font-size-base);
        }
      }
      .stacked-table-subtext {
        font-size: var(--bloom-font-size-sm);
        @media (min-width: $screen-md) {
          font-size: var(--bloom-font-size-xs);
        }
      }
    }
  }
}

.sale {
  .category-table {
    tr {
      td:before {
        font-size: var(--bloom-font-size-sm);

        @media (min-width: $screen-md) {
          font-size: var(--bloom-font-size-base);
        }
      }
      @media (min-width: $screen-md) {
        .stacked-table-header {
          width: 20%;


          &:nth-child(3) {
            width: 35%;
          }

          &:last-child {
            width: 25%;
          }
        }
      }
    }

    .stacked-table-header {
      .stacked-table-cell-container {
        .stacked-table-cell {
          font-size: var(--bloom-font-size-sm);

          @media (min-width: $screen-md) {
            font-size: var(--bloom-font-size-base);
          }
        }

        .stacked-table-subtext {
          font-size: var(--bloom-font-size-sm);

          @media (min-width: $screen-md) {
            font-size: var(--bloom-font-size-xs);
          }
        }
      }
    }
  }
}
