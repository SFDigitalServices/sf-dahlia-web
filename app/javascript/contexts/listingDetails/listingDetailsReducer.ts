import { createReducer } from "typesafe-actions"
import { ListingDetailsActions } from "./listingDetailsActions"
import { RailsUnit } from "../../api/types/rails/listings/RailsUnit"
import { RailsAmiChart } from "../../api/types/rails/listings/RailsAmiChart"
import { AmiChartData } from "./listingDetailsContext"

type ListingDetailsState = {
  fetchingUnits: boolean
  fetchedUnits: boolean
  fetchingAmiCharts: boolean
  fetchedAmiCharts: boolean
  units: RailsUnit[]
  amiChartData: AmiChartData
  amiCharts: RailsAmiChart[]
  fetchingAmiChartsError: Error
  fetchingUnitsError: Error
}

const ListingDetailsReducer = createReducer({ units: [] } as ListingDetailsState, {
  [ListingDetailsActions.StartFetchingUnits]: (state) => {
    return { ...state, fetchingUnits: true }
  },
  [ListingDetailsActions.StartFetchingAmiCharts]: (state) => {
    return { ...state, fetchingAmiCharts: true }
  },
  [ListingDetailsActions.FinishFetchingUnits]: (state, { payload: units }) => {
    const amiChartYears: Array<number> = []
    const amiChartTypes = []
    const amiChartPercentages = []

    units.forEach((unit: RailsUnit) => {
      const amiChartYear = amiChartYears.find((year) => {
        return year === unit.AMI_chart_year
      })

      const amiChartType = amiChartTypes.find((type) => {
        return type === unit.AMI_chart_type
      })

      const amiChartPercentage = amiChartPercentages.find((type) => {
        return type === unit.Max_AMI_for_Qualifying_Unit
      })

      if (!amiChartYear) {
        amiChartYears.push(unit.AMI_chart_year)
      }

      if (!amiChartType) {
        amiChartTypes.push(unit.AMI_chart_type)
      }

      if (!amiChartPercentage) {
        amiChartPercentages.push(unit.Max_AMI_for_Qualifying_Unit)
      }
    })

    return {
      ...state,
      fetchingUnits: false,
      fetchedUnits: true,
      units,
      amiChartData: {
        years: amiChartYears,
        percentages: amiChartPercentages,
        types: amiChartTypes,
      },
    }
  },
  [ListingDetailsActions.FinishFetchingAmiCharts]: (state, { payload: amiCharts }) => {
    return {
      ...state,
      fetchingAmiCharts: false,
      fetchedAmiCharts: true,
      amiCharts: amiCharts?.map((amiChart: RailsAmiChart) => {
        return {
          ...amiChart,
          chartType: amiChart?.values[0]?.chartType,
        }
      }),
    }
  },
})

export default ListingDetailsReducer
