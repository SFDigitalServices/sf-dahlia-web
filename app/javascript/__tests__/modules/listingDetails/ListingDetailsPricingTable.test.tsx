import React from "react"
import renderer from "react-test-renderer"
import { ListingDetailsPricingTable } from "../../../modules/listingDetails/ListingDetailsPricingTable"
import { closedRentalListing } from "../../data/RailsRentalListing/listing-rental-closed"
import { habitatListing } from "../../data/RailsSaleListing/listing-sale-habitat"
import { openSaleListing } from "../../data/RailsSaleListing/listing-sale-open"
import { units, unitsWithOneOccupancy } from "../../data/RailsListingUnits/listing-units"
import { amiCharts } from "../../data/RailsAmiCharts/ami-charts"
import ListingDetailsContext from "../../../contexts/listingDetails/listingDetailsContext"

describe("ListingDetailsPricingTable", () => {
  it("renders ListingDetailsPricingTable component with spinner before api call", () => {
    const tree = renderer.create(
      <ListingDetailsContext.Provider
        value={{
          units: [],
          amiCharts: [],
          fetchingUnits: true,
          fetchedUnits: false,
          fetchingAmiCharts: true,
          fetchedAmiCharts: false,
          fetchingAmiChartsError: null,
          fetchingUnitsError: null,
        }}
      >
        <ListingDetailsPricingTable listing={closedRentalListing} />
      </ListingDetailsContext.Provider>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it("renders ListingDetailsPricingTable component with rental listing", () => {
    const tree = renderer.create(
      <ListingDetailsContext.Provider
        value={{
          units,
          amiCharts,
          fetchingUnits: false,
          fetchedUnits: true,
          fetchingAmiCharts: false,
          fetchedAmiCharts: true,
          fetchingAmiChartsError: null,
          fetchingUnitsError: null,
        }}
      >
        <ListingDetailsPricingTable listing={closedRentalListing} />
      </ListingDetailsContext.Provider>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it("renders ListingDetailsPricingTable component with open sale listing", () => {
    const tree = renderer.create(
      <ListingDetailsContext.Provider
        value={{
          units,
          amiCharts,
          fetchingUnits: false,
          fetchedUnits: true,
          fetchingAmiCharts: false,
          fetchedAmiCharts: true,
          fetchingAmiChartsError: null,
          fetchingUnitsError: null,
        }}
      >
        <ListingDetailsPricingTable listing={openSaleListing} />
      </ListingDetailsContext.Provider>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it("renders ListingDetailsPricingTable component when rental listing", () => {
    const tree = renderer.create(
      <ListingDetailsContext.Provider
        value={{
          units,
          amiCharts,
          fetchingUnits: false,
          fetchedUnits: true,
          fetchingAmiCharts: false,
          fetchedAmiCharts: true,
          fetchingAmiChartsError: null,
          fetchingUnitsError: null,
        }}
      >
        <ListingDetailsPricingTable listing={closedRentalListing} />
      </ListingDetailsContext.Provider>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it("renders ListingDetailsPricingTable component when rental listing with AMI full text", () => {
    const tree = renderer.create(
      <ListingDetailsContext.Provider
        value={{
          units: unitsWithOneOccupancy,
          amiCharts,
          fetchingUnits: false,
          fetchedUnits: true,
          fetchingAmiCharts: false,
          fetchedAmiCharts: true,
          fetchingAmiChartsError: null,
          fetchingUnitsError: null,
        }}
      >
        <ListingDetailsPricingTable listing={closedRentalListing} />
      </ListingDetailsContext.Provider>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it("does not render ListingDetailsPricingTable when habitat listing", () => {
    const tree = renderer.create(
      <ListingDetailsContext.Provider
        value={{
          units,
          amiCharts,
          fetchingUnits: false,
          fetchedUnits: true,
          fetchingAmiCharts: false,
          fetchedAmiCharts: true,
          fetchingAmiChartsError: null,
          fetchingUnitsError: null,
        }}
      >
        <ListingDetailsPricingTable listing={habitatListing} />
      </ListingDetailsContext.Provider>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
