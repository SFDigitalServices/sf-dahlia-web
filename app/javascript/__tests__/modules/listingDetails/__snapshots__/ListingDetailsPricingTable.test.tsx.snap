// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingDetailsPricingTable does not render ListingDetailsPricingTable when habitat listing 1`] = `
<div
  className="md:pr-8 md:w-2/3 mx-2 w-full"
>
  <ul>
    <li>
      <p>
        1 person household: 51,300 to 61,300 per year
      </p>
    </li>
    <li>
      <p>
        2 people household: 58,600 to 68,600 per year
      </p>
    </li>
    <li>
      <p>
        3 people household: 65,950 to 75,950 per year
      </p>
    </li>
    <li>
      <p>
        4 people household: 73,250 to 83,250 per year
      </p>
    </li>
    <li>
      <p>
        5 people household: 79,100 to 99,100 per year
      </p>
    </li>
    <li>
      <p>
        6 people household: 85,000 to 95,000 per year
      </p>
    </li>
    <li>
      <p>
        7 people household: 90,850 to 100,850 per year
      </p>
    </li>
    <li>
      <p>
        8 people household: 96,700 to 106,700 per year
      </p>
    </li>
    <li>
      <p>
        9 people household: 102,600 to 112,600 per year
      </p>
    </li>
  </ul>
</div>
`;

exports[`ListingDetailsPricingTable renders ListingDetailsPricingTable component when rental listing 1`] = `
<div
  className="md:my-6 md:pr-8 sm:px-4 lg:px-0 md:w-2/3 px-2 w-full"
>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            1 person in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $100 to $5,941 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            2 people in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $7 to $6,550 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            3 people in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $7 to $6,329 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            4 people in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $7 to $6,937 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            5 people in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $7 to $11,666 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            6 people in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $10 to $7,916 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
</div>
`;

exports[`ListingDetailsPricingTable renders ListingDetailsPricingTable component with open sale listing 1`] = `
<div
  className="md:my-6 md:pr-8 sm:px-4 lg:px-0 md:w-2/3 px-2 w-full"
>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            1 person in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $100 to $5,941 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            2 people in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $7 to $6,550 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            3 people in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $7 to $6,329 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            4 people in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $7 to $6,937 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            5 people in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $7 to $11,666 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            6 people in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $10 to $7,916 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
</div>
`;

exports[`ListingDetailsPricingTable renders ListingDetailsPricingTable component with rental listing 1`] = `
<div
  className="md:my-6 md:pr-8 sm:px-4 lg:px-0 md:w-2/3 px-2 w-full"
>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            1 person in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $100 to $5,941 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            2 people in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $7 to $6,550 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            3 people in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $7 to $6,329 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            4 people in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $7 to $6,937 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            5 people in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $7 to $11,666 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
  <div
    className="mb-4"
  >
    <button
      aria-expanded={false}
      className="w-full text-left "
      data-test-id="content-accordion-button"
      onClick={[Function]}
    >
      <div
        className="flex justify-between mt-4  accordion-gray-theme__bar "
      >
        <span
          className="flex w-full justify-between items-center"
        >
          <span
            className="flex items-center"
          >
            6 people in household
          </span>
          <span
            className="flex items-center mr-2"
          >
            <div>
              Income $10 to $7,916 per month
            </div>
          </span>
        </span>
        <span
          className="ui-icon ui-base flex items-center"
          data-test-id="accordion-open"
        >
          <svg
            fill="#0077DA"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </div>
    </button>
  </div>
</div>
`;

exports[`ListingDetailsPricingTable renders ListingDetailsPricingTable component with spinner before api call 1`] = `
<div
  className="flex justify-center md:my-6 md:pr-8 md:px-0 md:w-2/3 px-3 w-full"
>
  <span
    className="ui-icon ui-large spinner-animation"
    data-test-id={null}
  >
    <svg
      fill="currentColor"
      viewBox="0 0 100 100"
    >
      <circle
        cx="50"
        cy="50"
        fill="none"
        r="32"
        stroke="currentColor"
        strokeDasharray="150.79644737231007 52.26548245743669"
        strokeWidth="9"
      />
    </svg>
  </span>
</div>
`;
